src/js/components/CookieGuard.tsx

```diff
@@ -0,0 +1,113 @@
+import { FC, Fragment, useState } from 'react';
+import { Dialog, Disclosure, Transition } from '@headlessui/react';
+import { useCookies } from '../hooks/useCookies';
+
+type CookiePopupProps = {
+    title: string;
+    beforeOptions: JSX.Element | string;
+};
+
+export const CookieGuard: FC<CookiePopupProps> = ({ title, beforeOptions }) => {
+    let [isOpen, setIsOpen] = useState(true);
+    const { cookies, setCookies } = useCookies();
+
+    const [analytics, setAnalytics] = useState(cookies.analytics);
+    const [marketing, setMarketing] = useState(cookies.marketing);
+
+    const onSetCookies = (cookies: {
+        analytics: boolean;
+        marketing: boolean;
+    }) => {
+        setCookies(cookies);
+        setIsOpen(false);
+    };
+
+    return (
+        <Dialog
+            open={isOpen}
+            onClose={() => {}}
+            style={{ position: 'relative', zIndex: 999 }}
```

Comment:
Wat betreft styling ga je hier wel wat door elkaar zie inline styles, classnames en tailwind.
Volgens mij ben je hier nog mee bezig maar zou kijken naar hoe je als gebruiker de default styles zo eenvoudig mogelijk kan overschrijven of je eigen toe voegen. Mijn idee was om standaard een css file te builden met classnames (dus geen css module) en dat je eventueel dat bestand ook niet kunt inladen maar je eigen styles kan toepassen maar dan heb je wel direct een overzicht van alle beschikbare classes.

Reply:
Styling heb ik niks aan gedaan ;).

Was eigenlijk als draft PR bedoeld ahah